common/gera/map_test.go

```diff
@@ -0,0 +1,626 @@
+/*
+ * === This file is part of ALICE O² ===
+ *
+ * Copyright 2020-2024 CERN and copyright holders of ALICE O².
+ * Author: Teo Mrnjavac <teo.mrnjavac@cern.ch>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package gera
+
+import (
+	"maps"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"gopkg.in/yaml.v3"
+)
+
+var _ = Describe("hierarchical key-value store", func() {
```

Comment:
Is there some code coverage report generated? It would be nice to see whether you really test everything, because I cannot tell if everything is tested, from simple read through

Reply:
`ginkgo -r -v -race --trace --coverpkg=github.com/AliceO2Group/Control/common/gera --coverprofile=.coverage-report.out`
...
```
Ran 21 of 21 Specs in 0.023 seconds
SUCCESS! -- 21 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS
coverage: 80.2% of statements in github.com/AliceO2Group/Control/common/gera
composite coverage: 80.2% of statements
```

Reply:
Or actually, just `make coverage`. AFAICT you added the command this April, and thanks for that!

Reply:
oh heck... you are right, I already forgot about it :laughing:  I took a look at the coverage and it looks fine. However, I did not see any error path tests. But I trust you, that it is not necessary.